### PR TITLE
Feature/27 support root session

### LINKED ISSUE #27: Support the RootSession
Create the RootApplicationFactory and make RootSession available from ApplicationManager. 
Allow to search elements from the root session without switching the app. Probably should be either the default behavior for AutomationId locator type, or the parameter in the constructor/factory method.
Remove the logging from the application constructor for the cases like the RootSession
Implement a factory for attaching to running application via TopWindow handle

### PR DESCRIPTION
**Pull request by mialeska**
Closes #27 
Inject root session into the application. update factories and add WindowHandleApplicationFactory
Implemented ProcessManager and WinAppDriverLauncher
add ProcessManagerTests and WindowHandleApplicationFactory
Window will be reworked to implement the search from the root session context withtin the issue #24

### CODE REVIEW

File: Aquality.WinAppDriver/src/Aquality.WinAppDriver/Utilities/IWinAppDriverLauncher.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        /// Attepmts to find running WinAppDriver process; if no process found, starts it
+        /// </summary>
+        /// <returns>Instance of process if it was started or null if it wasn't.</returns>
+        Process StartWinAppDriverIfRequired();
+
+        /// <summary>
+        /// Attempts to stop WinAppDriver process
+        /// </summary>
+        /// <returns>True if the process was found and stopped, false otherwise</returns>
+        bool TryToStopWinAppDriver();
```
Comment (paveliam) [inline Aquality.WinAppDriver/src/Aquality.WinAppDriver/Utilities/IWinAppDriverLauncher.cs:20]: Should this method be public? Maybe it makes sense to remove `TryTo` from its name?

File: Aquality.WinAppDriver/src/Aquality.WinAppDriver/Applications/WindowHandleApplicationFactory.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            this.driverServerUri = driverServerUri;
+            this.getWindowHandleFunction = getWindowHandleFunction;
+            isRemote = serviceProvider.GetRequiredService<IApplicationProfile>().IsRemote;
+        }
+
+        public override Application Application
+        {
+            get
+            {
+                LocalizedLogger.Info(isRemote ? "loc.application.driver.service.remote" : "loc.application.driver.service.local.start", driverServerUri);
```
Comment (paveliam) [inline Aquality.WinAppDriver/src/Aquality.WinAppDriver/Applications/WindowHandleApplicationFactory.cs]: please create variable for `isRemote ? "loc.application.driver.service.remote" : "loc.application.driver.service.local.start"`

Comment (paveliam) [reply Aquality.WinAppDriver/src/Aquality.WinAppDriver/Applications/WindowHandleApplicationFactory.cs]: Maybe it makes sense to leave only one message? Probably `driverServerUri` has different values for remote and local session and we can identify it by its value.

**Review by paveliam — APPROVED** [review]

File: Aquality.WinAppDriver/src/Aquality.WinAppDriver/Utilities/IWinAppDriverLauncher.cs (hunk 1 above)
Comment (mialeska) [reply Aquality.WinAppDriver/src/Aquality.WinAppDriver/Utilities/IWinAppDriverLauncher.cs:20]: if the WinAppDriver is not stopped, the returned result would be false.
Should the method called `StopWinAppDriver` return boolean value?

File: Aquality.WinAppDriver/src/Aquality.WinAppDriver/Applications/WindowHandleApplicationFactory.cs (hunk 2 above)
Comment (mialeska) [reply Aquality.WinAppDriver/src/Aquality.WinAppDriver/Applications/WindowHandleApplicationFactory.cs]: actually it is configurable and could be the same

**Review by knysh — APPROVED** [review]

File: Aquality.WinAppDriver/src/Aquality.WinAppDriver/Applications/WindowHandleApplicationFactory.cs (hunk 2 above)
Comment (mialeska) [reply Aquality.WinAppDriver/src/Aquality.WinAppDriver/Applications/WindowHandleApplicationFactory.cs]: comment#1 fixed

**Pull request merged**